test_runner/performance/test_perf_ingest_using_pgcopydb.py

```diff
@@ -0,0 +1,343 @@
+                    duration_match = re.search(r"\d+h\d+m|\d+s|\d+ms|\d+\.\d+s", line)
+                    if duration_match:
+                        duration_str = duration_match.group(0)
+                        duration_seconds = convert_to_seconds(duration_str)
+                        metrics[metric_name] = duration_seconds
+
+    # Report metrics to the database
+    report_metrics_to_db(metrics)
+
+
+def get_endpoint_id():
+    """Extracts and returns the first segment of the hostname from the PostgreSQL URI stored in BENCHMARK_INGEST_TARGET_CONNSTR."""
+    connstr = os.getenv("BENCHMARK_INGEST_TARGET_CONNSTR")
+    if connstr is None:
+        raise OSError("BENCHMARK_INGEST_TARGET_CONNSTR environment variable is not set.")
+
+    # Parse the URI
+    parsed_url = urlparse(connstr)
+
+    # Extract the hostname and split to get the first segment
+    hostname = parsed_url.hostname
+    if hostname is None:
+        raise ValueError("Unable to parse hostname from BENCHMARK_INGEST_TARGET_CONNSTR")
+
+    # Split the hostname by dots and take the first segment
+    endpoint_id = hostname.split(".")[0]
+
+    return endpoint_id
+
+
+def report_metrics_to_db(metrics):
+    """Inserts parsed metrics into the performance database."""
+    # Connection string for the performance database
+    connstr = os.getenv("PERF_TEST_RESULT_CONNSTR")
+    if not connstr:
+        raise OSError("PERF_TEST_RESULT_CONNSTR environment variable is not set.")
+    commit_hash = os.getenv("COMMIT_HASH")
+    if not commit_hash:
+        raise OSError("COMMIT_HASH environment variable is not set.")
+    endpoint_id = get_endpoint_id()
+    project_type = os.getenv("TARGET_PROJECT_TYPE")
+    if not project_type:
+        raise OSError("TARGET_PROJECT_TYPE environment variable is not set.")
+    platform = f"pg16-{project_type}-us-east-2-staging"
+
+    # Connect to the database
+    with psycopg2.connect(connstr) as conn:
+        with conn.cursor() as cur:
+            for metric_name, metric_value in metrics.items():
+                cur.execute(
+                    """
+                    INSERT INTO public.perf_test_results (suit, revision, platform, metric_name, metric_value, metric_unit, metric_report_type, recorded_at_timestamp, label_1)
+                    VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)
+                """,
+                    (
+                        "pgcopydb_ingest_bench",  # Suit
+                        commit_hash,  # Revision (example value, replace as needed)
+                        platform,
+                        metric_name,
+                        metric_value,  # in seconds
+                        "seconds",  # Metric unit
+                        "lower_is_better",  # Metric report type
+                        datetime.now(),  # Recorded timestamp
+                        endpoint_id,  # Label 1
+                    ),
+                )
+            conn.commit()
+
+
+@pytest.fixture
```

Comment:
Fixtures by default are function-scoped, this can be a session one: `@pytest.fixture(scope="session")`

Reply:
ok, changed it, however doesn't make a difference as we have only one function, one package, one module, one session in this test